aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move

```diff
@@ -1,170 +1,324 @@
+
+        debug::print(&val);
+        debug::print(&std::string::utf8(b"before42after"));
+        assert!(val == std::string::utf8(b"before42after"), 5);
     }
 
     public entry fun verify_string_snapshot_concat() {
         let snapshot = aggregator_v2::create_snapshot(std::string::utf8(b"42"));
         let snapshot2 = aggregator_v2::string_concat(std::string::utf8(b"before"), &snapshot, std::string::utf8(b"after"));
-        assert!(aggregator_v2::read_snapshot(&snapshot2) == std::string::utf8(b"before42after"), 0);
+        assert!(aggregator_v2::read_snapshot(&snapshot2) == std::string::utf8(b"before42after"), 6);
     }
 
+    fun init<Agg: store>(account: &signer, use_type: u32) {
+        if (use_type == USE_RESOURCE_TYPE) {
+            move_to(account, AggregatorInResource<Agg> { data: vector::empty() });
+        } else if (use_type == USE_TABLE_TYPE) {
+            move_to(account, AggregatorInTable<Agg> { data: table::new() });
+        } else if (use_type == USE_RESOURCE_GROUP_TYPE) {
+            move_to(account, AggregatorInResourceGroup<Agg> { data: vector::empty() });
+        } else {
+            assert!(false, EINVALID_ARG);
+        };
+    }
 
-    /// Checks that the ith aggregator has expected value. Useful to inject into
-    /// transaction block to verify successful and correct execution.
-    public entry fun check(account: &signer, i: u64, expected: u128) acquires AggregatorStore {
+    public entry fun init_aggregator<Element: store>(account: &signer, use_type: u32) {
+        init<Aggregator<Element>>(account, use_type);
+    }
+
+    public entry fun init_snapshot<Element: store>(account: &signer, use_type: u32) {
+        init<AggregatorSnapshot<Element>>(account, use_type);
+    }
+
+    fun insert<Agg: store>(account_addr: address, use_type: u32, i: u64, e: Agg) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
+        assert!(use_type == USE_RESOURCE_TYPE || use_type == USE_TABLE_TYPE || use_type == USE_RESOURCE_GROUP_TYPE, EINVALID_ARG);
+
+        let vector_data = if (use_type == USE_RESOURCE_TYPE) {
+            &mut borrow_global_mut<AggregatorInResource<Agg>>(account_addr).data
+        } else if (use_type == USE_TABLE_TYPE) {
+            let data = &mut borrow_global_mut<AggregatorInTable<Agg>>(account_addr).data;
+            let outer = i / 10;
+            let inner = i % 10;
+            i = inner;
+            if (!table::contains(data, outer)) {
+                table::add(data, outer, vector::empty());
+            };
+
+            table::borrow_mut(data, outer)
+        } else { // if (use_type == USE_RESOURCE_GROUP_TYPE) {
+            &mut borrow_global_mut<AggregatorInResourceGroup<Agg>>(account_addr).data
+        };
+
+        if (vector::length(vector_data) == i) {
+            vector::push_back(vector_data, option::some(e));
+        } else {
+            let option_data = vector::borrow_mut(vector_data, i);
```

Comment:
This will give out of bounds if i > length?

Reply:
yes, that's fine for current tests (you need to add them sequentially. we can change later if needed.